### PR TITLE
fix: avoid dialing/listening on dns addresses

### PR DESCRIPTION
**Pull request by Stebalien**
See https://github.com/libp2p/go-libp2p/issues/841

### CODE REVIEW

File: transport.go [hunk 1]
```diff
@@ -190,9 +190,12 @@ func (t *transport) Dial(ctx context.Context, raddr ma.Multiaddr, p peer.ID) (tp
 	}, nil
 }
 
+// Don't use mafmt.QUIC as we don't want to dial DNS addresses. Just /ip{4,6}/udp/quic
```
Comment (Stebalien) [inline transport.go:193]: @marten-seemann ^^

Comment (marten-seemann) [reply transport.go:193]: 🤦‍♂️

**Comment by Stebalien** [comment]
I completely forgot about that PR.

File: transport.go (hunk 1 above)
Comment (Stebalien) [reply transport.go:193]: Yeah, it's confusing. We may want to just go ahead and revert the change in mafmt (we changed it from IP only to IP+DNS). But it does make sense to allow DNS there, just not _here_.

**Pull request merged**